src/shell/dotnet/src/Shell/Fdc3/ComposeUIHostManifestMapper.cs

```diff
@@ -43,39 +44,89 @@ public ModuleDetails MapModuleDetails(Fdc3App fdc3App)
         return fdc3App.Type switch
         {
             AppType.Web => MapWebManifestDetails(fdc3App),
+            AppType.Native => MapNativeManifestDetails(fdc3App),
             _ => throw new NotSupportedException($"The {fdc3App.Type} is currently not supported for hostmanifest conversion!"),
         };
     }
 
+    private ModuleDetails MapNativeManifestDetails(Fdc3App fdc3App)
```

Comment:
Consider creating an interface of abstract class for this functionality and implement it for different manifest types